src/engine/source/indexerconnector/src/indexerConnector.cpp

```diff
@@ -81,26 +81,54 @@ static void initConfiguration(SecureCommunication& secureCommunication, const nl
         .caRootCertificate(caRootCertificate);
 }
 
-static void builderBulkDelete(std::string& bulkData, std::string_view id, std::string_view index)
+void IndexerConnector::builderBulkDelete(std::string& bulkData, std::string_view id, std::string_view index)
 {
-    bulkData.append(R"({"delete":{"_index":")");
-    bulkData.append(index);
-    bulkData.append(R"(","_id":")");
-    bulkData.append(id);
-    bulkData.append(R"("}})");
-    bulkData.append("\n");
+    bulkData.append(QueryBuilder::builder()
+                        .openObject()
+                        .addField("delete")
+                        .addNestedFieldAndValue("_index", index)
+                        .objectSeparator()
+                        .addFieldAndValue("_id", id)
+                        .closeObject()
+                        .closeObject()
+                        .addNewLine()
+                        .build());
 }
 
-static void builderBulkIndex(std::string& bulkData, std::string_view id, std::string_view index, std::string_view data)
+void IndexerConnector::builderBulkIndex(std::string& bulkData,
+                                        std::string_view id,
+                                        std::string_view index,
+                                        std::string_view data)
 {
-    bulkData.append(R"({"index":{"_index":")");
-    bulkData.append(index);
-    bulkData.append(R"(","_id":")");
-    bulkData.append(id);
-    bulkData.append(R"("}})");
-    bulkData.append("\n");
-    bulkData.append(data);
-    bulkData.append("\n");
+
+    if (id.empty())
```

Comment:
This conditional could be a part of the build() method.

Reply:
According to the discussion above, the approach will be changed.